VideoWeb/VideoWeb/Mappings/ClaimsPrincipalToUserProfileResponseMapper.cs

```diff
@@ -17,36 +17,33 @@ public UserProfileResponse Map(ClaimsPrincipal user)
             FirstName = user.Claims.FirstOrDefault(c => c.Type == ClaimTypes.GivenName)?.Value,
             LastName = user.Claims.FirstOrDefault(c => c.Type == ClaimTypes.Surname)?.Value,
             DisplayName = user.Claims.Last(c => c.Type == ClaimTypes.Name).Value,
-            Username = user.Identity.Name.ToLower().Trim(),
+            Username = user.Identity?.Name?.ToLower().Trim(),
             Name =  user.Claims.Last(c => c.Type == ClaimTypes.Name).Value
         };
         return response;
     }
-
+    
     private static List<Role> DetermineRolesFromClaims(ClaimsPrincipal user)
     {
-        var roles = new List<Role>();
+        var roleChecks = new Dictionary<Role, Func<ClaimsPrincipal, bool>>
+        {
+            { Role.VideoHearingsOfficer, u => u.IsInRole(AppRoles.VhOfficerRole) },
```

Comment:
Sonar complained about the rising cognitive complexity with an additional is statement, so I refactored this into a dictionary